### PR TITLE
[PIE-1385] Update API payload to contain name and version

### PR DESCRIPTION
**Pull request by KatieWright26**
This PR appends collector name and version to the API payload for the test-collector per [PIE-1385](https://linear.app/buildkite/issue/PIE-1385/update-net-collector-api-payload-to-pass-collector-name)

### CODE REVIEW

File: Buildkite.TestAnalytics.Common/Payload.fs [hunk 1]
```diff
@@ -44,7 +46,9 @@ module Payload =
                   Branch = optionalEnvVar "BUILDKITE_BRANCH"
                   CommitSha = optionalEnvVar "BUILDKITE_COMMIT"
                   Message = optionalEnvVar "BUILDKITE_MESSAGE"
-                  Url = optionalEnvVar "BUILDKITE_BUILD_URL" }
+                  Url = optionalEnvVar "BUILDKITE_BUILD_URL"
+                  Version = "0.1.1"
```
Comment (KatieWright26) [inline Buildkite.TestAnalytics.Common/Payload.fs:50]: I really want to extract this value from the .fsproj file, but can't quite see the correct way. I've tried things like `System.Reflection.AssemblyVersionAttribute` and `Assembly.GetEntryAssembly().GetName().Version` but no luck...

**Comment by KatieWright26** [comment]
@jimsynz I have a few questions as well as my one about extracting the Version from `Buildkite.TestAnalytics.Common.fsproj`:
- Is there a script to bump the version in the above file that's not just manually changing it in the code?
- Does the version in `Buildkite.TestAnalytics.Common.fsproj` need to match that in `Buildkite.TestAnalytics.Xunit`

**Comment by jimsynz** [comment]
> @jimsynz I have a few questions as well as my one about extracting the Version from `Buildkite.TestAnalytics.Common.fsproj`:
> 
>     * Is there a script to bump the version in the above file that's not just manually changing it in the code?

I'm not sure I'm afraid. There _should_ be a way to extract the version of the current project but I don't know what it is. I've asked @mbuhot to swing by and see if he knows the answers.

>     * Does the version in `Buildkite.TestAnalytics.Common.fsproj` need to match that in `Buildkite.TestAnalytics.Xunit`

No, they're separate nuget packages, so they can be versioned independently of each other.

**Comment by mbuhot** [comment]
```fsharp
Version = System.Reflection.Assembly.GetExecutingAssembly().GetName().Version.ToString()
```

Should give you the version number of the `Buildkite.TestAnalytics.Common` assembly at runtime.

**Review by niceking — APPROVED** [review]
Looks sane to me and there are tests, but I also don't know dotnet so maybe an approving review from someone else in the team would also help!

**Comment by KatieWright26** [comment]
Thanks @mbuhot , @jimsynz are you able to do another review?

**Review by jimsynz — APPROVED** [review]
LGTM

**Comment by KatieWright26** [comment]
@jimsynz are you able to do the release on NuGet? I can't sign up under my buildkite creds and it looks like you're the owner :)

**Comment by jimsynz** [comment]
@KatieWright26 0.1.2 has been published.

**Pull request merged**